be/src/formats/parquet/encoding_dict.h

```diff
@@ -81,9 +82,67 @@ class DictEncoder final : public Encoder {
     faststring _buffer;
 };
 
+class CacheAwareDictDecoder : public Decoder {
+public:
+    CacheAwareDictDecoder() { _set_dict_size_threshold(); }
+    ~CacheAwareDictDecoder() override = default;
+
+    Status next_batch(size_t count, ColumnContentType content_type, Column* dst, const FilterData* filter) override {
+        switch (content_type) {
+        case DICT_CODE: {
+            FixedLengthColumn<int32_t>* data_column;
+            if (dst->is_nullable()) {
+                auto nullable_column = down_cast<NullableColumn*>(dst);
+                nullable_column->null_column()->append_default(count);
+                data_column = down_cast<FixedLengthColumn<int32_t>*>(nullable_column->data_column().get());
+            } else {
+                data_column = down_cast<FixedLengthColumn<int32_t>*>(dst);
+            }
+            size_t cur_size = data_column->size();
+            data_column->resize_uninitialized(cur_size + count);
+            int32_t* __restrict__ data = data_column->get_data().data() + cur_size;
+            auto decoded_num = _rle_batch_reader.GetBatch(reinterpret_cast<uint32_t*>(data), count);
+            if (decoded_num < count) {
+                return Status::InternalError("didn't get enough data from dict-decoder");
+            }
+            break;
+        }
+        case VALUE: {
+#ifdef BE_TEST
+            return _next_batch_value(count, dst, filter);
+#else
+            if (_get_dict_size() > _dict_size_threshold && config::parquet_cache_aware_dict_decoder_enable) {
+                return _next_batch_value(count, dst, filter);
+            } else {
+                return _next_batch_value(count, dst, nullptr);
+            }
+#endif // BE_TEST
+        }
+        default:
+            return Status::NotSupported("read type not supported");
+        }
+        return Status::OK();
+    }
+
+protected:
+    virtual size_t _get_dict_size() const = 0;
+    virtual Status _next_batch_value(size_t count, Column* dst, const FilterData* filter) = 0;
+    RleBatchDecoder<uint32_t> _rle_batch_reader;
+
+private:
+    static const size_t DEFAULT_L2_CACHE_SIZE = 1 * 1024 * 1024;
+    void _set_dict_size_threshold() {
+        const auto& cache_sizes = CpuInfo::get_cache_sizes();
+        auto l2_cache_size = cache_sizes[CpuInfo::L2_CACHE];
```

Comment:
implement one function CpuInfo::get_l2_cache_size like 'static long get_l3_cache_size() `?

Reply:
ok, will do it.